prisma/schema.prisma

```diff
@@ -0,0 +1,175 @@
+  provider = "node node_modules/prisma-nestjs-graphql"
+  output = "../src/@generated"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+// Represents possible genders.
+enum Gender {
+  // Represents the male gender.
+  MALE
+
+  // Represents the female gender.
+  FEMALE
+
+  // Represents non-binary genders.
+  NON_BINARY
+}
+
+// It represents the status of the anime, in terms of whether it is on air, finished, upcoming, etc.
+enum AnimeStatus {
+  // Anime is currently on air.
+  ONGOING
+
+  // Anime has finished airing.
+  FINISHED
+
+  // Anime is yet to be aired.
+  UPCOMING
+
+  // Anime is on an indefinite break
+  HIATUS
+
+  // Anime is in a temporary and definite break
+  ON_BREAK
+
+  // Anime was cancelled/abandoned suddenly
+  DROPPED
+}
+
+// Represents a user or profile.
+model User {
+  id             Int         @id @unique @default(autoincrement()) // Unique identifier for a user.
+  email          String      @unique @db.VarChar(255) // Email address of the user.
+  username       String      @unique @db.VarChar(12) // Username chosen by the user.
+  avatarURL      String // URL to the user's avatar image.
+  gender         Gender // Gender of the user, as defined in the Gender enum.
+  followersCount Int        @default(0) // Number of followers of the user.
+  followedCount  Int        @default(0) // Number of users the user is following.
+  animesCount    Int        @default(0) // Number of animes liked by the user
+  reviewsCount   Int        @default(0) // Number of reviews the user has written.
+  favoriteGenres UserGenre[] // List of genres favored by the user.
+  favoriteAnimes UserAnime[] // List of animes favored by the user.
+  followed       Follower[]  @relation("UserFollows") // List of users that this user is following.
+  followers      Follower[]  @relation("UserFollowedBy") // List of users who are following this user.
+  reviews        Review[] // List of reviews written by the user.
+
+  createdAt DateTime @default(now()) // Date and time when was created
+  updatedAt DateTime @updatedAt // Date and time when was last updated
+
+  @@index([id]) // Index for enhancing search performance.
+}
+
+// Intermediate model for connecting the followers/following of an individual user
+model Follower {
+  id         Int  @id @default(autoincrement())
+  followedId Int
+  followerId Int
+  followed   User @relation("UserFollows", fields: [followedId], references: [id])
+  follower   User @relation("UserFollowedBy", fields: [followerId], references: [id])
+
+  createdAt DateTime @default(now()) // Date and time when was created
+
+  @@index([id, followedId, followerId])
+}
+
+// Represents a genre of anime.
+model Genre {
+  id     Int         @id @default(autoincrement()) // Unique identifier for a genre.
+  name   String // Name of the genre.
+  animes Anime[] // List of animes associated with this genre.
+  fans   UserGenre[] // List of users who are fans of this genre.
```

Comment:
Ohhh yes now I see it, this is a intermediary table, we will never directly perform a query against this, this will only serve as the link between both tables. Good catch.